Compilation Syntax Error with C++ 20

When strictly compiling with C++20 then the build fails with Error `C2760` with Visual Studio. Visual studio does not accept the syntax in `cyclonedds-cxx\src\ddscxx\include\dds\topic\detail\TopicInstanceImpl.hpp(42,34)`.


```bash
Build FAILED.

"C:\Users\sven\Downloads\test\cyclonedds-cxx\build\ALL_BUILD.vcxproj" (default target) (1) ->
"C:\Users\sven\Downloads\test\cyclonedds-cxx\build\examples\roundtrip\cxxRoundtripPing.vcxproj" (default target) (3) ->
(ClCompile target) ->
  C:\Users\sven\Downloads\test\cyclonedds-cxx\src\ddscxx\include\dds\topic\detail\TopicInstanceImpl.hpp(42,34): error C2760: syntax error: '::' was unexpected here; expected 'id-expression' [C:\Users\sven\Down
loads\test\cyclonedds-cxx\build\examples\roundtrip\cxxRoundtripPing.vcxproj]
  C:\Users\sven\Downloads\test\cyclonedds-cxx\src\ddscxx\include\dds\topic\detail\TopicInstanceImpl.hpp(42,47): error C2270: 'InstanceHandle': modifiers not allowed on nonmember functions [C:\Users\sven\Downlo
ads\test\cyclonedds-cxx\build\examples\roundtrip\cxxRoundtripPing.vcxproj]
  C:\Users\sven\Downloads\test\cyclonedds-cxx\src\ddscxx\include\dds\topic\detail\TopicInstanceImpl.hpp(42,47): error C2955: 'dds::topic::TopicInstance': use of class template requires template argument list [
C:\Users\sven\Downloads\test\cyclonedds-cxx\build\examples\roundtrip\cxxRoundtripPing.vcxproj]
  C:\Users\sven\Downloads\test\cyclonedds-cxx\src\ddscxx\include\dds\topic\detail\TopicInstanceImpl.hpp(42,47): error C2063: 'dds::core::InstanceHandle': not a function [C:\Users\sven\Downloads\test\cyclonedds
-cxx\build\examples\roundtrip\cxxRoundtripPing.vcxproj]
  C:\Users\sven\Downloads\test\cyclonedds-cxx\src\ddscxx\include\dds\topic\detail\TopicInstanceImpl.hpp(42,47): error C2226: syntax error: unexpected type 'dds::core::InstanceHandle' [C:\Users\sven\Downloads\t
est\cyclonedds-cxx\build\examples\roundtrip\cxxRoundtripPing.vcxproj]
```

How to reproduce?
Compile Cyclonedds-Cxx with `/std:c++20`  in Visual Studio.

--- a/src/ddscxx/include/dds/topic/detail/TopicInstanceImpl.hpp
+++ b/src/ddscxx/include/dds/topic/detail/TopicInstanceImpl.hpp
@@ -39,12 +39,6 @@ TopicInstance<T>::TopicInstance(const ::dds::core::InstanceHandle& h, const T& s
     : h_(h), sample_(sample) { }
 
 template <typename T>
-TopicInstance<T>::operator const ::dds::core::InstanceHandle() const
-{
-    return h_;
-}
-
-template <typename T>
 const ::dds::core::InstanceHandle TopicInstance<T>::handle() const
 {
     return h_;
@@ -75,6 +69,12 @@ void TopicInstance<T>::sample(const T& sample)
 }
 
 }
+}
+
+template <typename T>
+dds::topic::TopicInstance<T>::operator const ::dds::core::InstanceHandle() const
+{
+    return h_;
 }
 
 // End of implementation

